src/main/java/ch/njol/skript/effects/EffPlaySound.java

```diff
@@ -19,166 +19,242 @@
-		"",
-		"Please note that sound names can get changed in any Minecraft or Spigot version, or even removed from Minecraft itself."})
+@Description({
+	"Plays a sound at given location for everyone or just for given players, or plays a sound to specified players. " +
+	"Both Minecraft sound names and " +
+	"<a href=\"https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Sound.html\">Spigot sound names</a> " +
+	"are supported. Playing resource pack sounds are supported too. The sound category is 'master' by default. ",
+	"",
+	"Playing a sound from an entity directly will result in the sound coming from said entity, even while moving.",
+	"If the sound is custom, a location emitter will follow the entity. Do note that pitch and volume ",
+	"are reflected based on the entity, and Minecraft may not use the values from this syntax.",
+	"",
+	"If using Paper 1.19.4+ or Adventure API 4.12.0+ you can utilize sound seeds. Minecraft sometimes have a set of sounds under one sound ID ",
+	"that will randomly play, to counter this, you can directly state which seed to use.",
+	"",
+	"Please note that sound names can get changed in any Minecraft or Spigot version, or even removed from Minecraft itself.",
+})
 @Examples({
 	"play sound \"block.note_block.pling\" # It is block.note.pling in 1.12.2",
 	"play sound \"entity.experience_orb.pickup\" with volume 0.5 to the player",
-	"play sound \"custom.music.1\" in jukebox category at {speakerBlock}"
+	"play sound \"custom.music.1\" in jukebox category at {speakerBlock}",
+	"play sound \"BLOCK_AMETHYST_BLOCK_RESONATE\" with seed 1 on target entity for the player #1.20.1+"
 })
-@Since("2.2-dev28, 2.4 (sound categories)")
+@RequiredPlugins("Paper 1.19.4+ or Adventure API 4.12.0+ (sound seed)")
+@Since("2.2-dev28, 2.4 (sound categories), INSERT VERSION (sound seed & entity emitter)")
 public class EffPlaySound extends Effect {
 
-	private static final Pattern KEY_PATTERN = Pattern.compile("([a-z0-9._-]+:)?[a-z0-9/._-]+");
-	
+	private static final boolean ADVENTURE_API = Skript.classExists("net.kyori.adventure.sound.Sound$Builder");
+	public static final Pattern KEY_PATTERN = Pattern.compile("([a-z0-9._-]+:)?[a-z0-9/._-]+");
+
 	static {
+		String additional = "";
+		if (ADVENTURE_API)
+			additional = "[[with] seed %-number%] ";
 		Skript.registerEffect(EffPlaySound.class,
-				"play sound[s] %strings% [(in|from) %-soundcategory%] " +
-						"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] at %locations% [(to|for) %-players%]",
-				"play sound[s] %strings% [(in|from) %-soundcategory%] " +
-						"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]"
+				"play sound[s] %strings% " + additional + "[(in|from) %-soundcategory%] " +
+						"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] (at|on|from) %locations/entities% [(to|for) %-players%]",
```

Comment:
What is the `on` pattern for?

Reply:
for usage like `play sound ... on the last spawned entity` because the entity can come directly from an entity.

Reply:
Will it play from on top of it? Above it? Or is it just the same as from the entity?

Reply:
See the video. It makes the audio emit from the entity. It'll be the pivot location of the entity. See video linked for how the sound emits.

Reply:
Ah so it's from it, the syntax saying "on" confused me and I wasn't sure if it was supposed to be some extra feature.